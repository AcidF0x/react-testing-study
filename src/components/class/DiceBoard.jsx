import React from "react";
import Dice from "../../services/Dice";

class DiceRoller extends React.Component {
  state = {
    currentDice: null
  }

  rollDice = () => {
    const currentNumber = Dice.roll();
    this.setState({
      currentDice: currentNumber
    })
    this.props.diceRolled(currentNumber)
  }

  render() {
    return (
      <div className="dice-right-panel">
        <div className="dice-result">
          {
            this.state.currentDice !== null &&
            <h4>
              { this.state.currentDice }이/가 나왔습니다 !!
            </h4>
          }
        </div>
        <div className="dice-desc">
          <button onClick={this.rollDice}>던지기</button>
        </div>
      </div>
    )
  }
}

export default DiceRoller;
