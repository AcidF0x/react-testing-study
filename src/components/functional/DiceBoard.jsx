import React, {useState} from "react";
import Dice from "../../services/Dice";

const DiceBoard = ({diceRolled}) => {
  const [currentDice, setCurrentDice] = useState(null);
  const rollDice = () => {
    const currentNumber = Dice.roll()
    setCurrentDice(currentNumber)
    diceRolled(currentNumber)
  }

  return (
    <div className="dice-right-panel">
      <div className="dice-result">
        {
          currentDice !== null &&
          <h4>
            { currentDice }가 나왔습니다 !!
          </h4>
        }
      </div>
      <div className="dice-desc">
        <button onClick={rollDice}>던지기</button>
      </div>
    </div>
  )
}
export default DiceBoard;
