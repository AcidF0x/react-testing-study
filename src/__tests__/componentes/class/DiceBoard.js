import React from "react";
import DiceBoard from "../../../components/class/DiceBoard";
import { mount } from "enzyme";
import Dice from "../../../services/Dice";
jest.mock('../../../services/Dice');

describe('DiceBoard 테스트', () => {
  it('렌더링을 할수있다', () => {
    // Given
    // When
    const wrapper = mount(<DiceBoard/>);

    // Then
    expect(wrapper).toMatchSnapshot();
  });
  it('Dice.Roll을 실행 한다.', () => {
    // Given
    const diceRolled = jest.fn();
    const wrapper = mount(<DiceBoard diceRolled={diceRolled}/>);
    // When
    wrapper.instance().rollDice()

    // Then
    expect(Dice.roll).toHaveBeenCalled()
  })

  it('Dice.Roll을 실행 한다.', () => {
    // Given
    const diceRolled = jest.fn();
    const wrapper = mount(<DiceBoard diceRolled={diceRolled}/>);
    const button = wrapper.find('button')

    // When
    button.simulate('click')

    // Then
    expect(Dice.roll).toHaveBeenCalled()
  })

  it('결과 값을 state에 저장 한다.', () => {
    // Given
    const mockDiceNumber = 2;
    Dice.roll.mockReturnValueOnce(mockDiceNumber);

    const diceRolled = jest.fn();
    const wrapper = mount(<DiceBoard diceRolled={diceRolled}/>);
    // When
    wrapper.instance().rollDice()

    // Then
    expect(wrapper.state().currentDice).toBe(mockDiceNumber)
  })

  it('diceRolled 실행 한다.', () => {
    // Given
    const mockDiceNumber = 2;
    Dice.roll.mockReturnValueOnce(mockDiceNumber);

    const diceRolled = jest.fn();
    const wrapper = mount(<DiceBoard diceRolled={diceRolled}/>);
    // When
    wrapper.instance().rollDice()

    // Then
    expect(diceRolled).toHaveBeenCalledWith(mockDiceNumber)
  })
});