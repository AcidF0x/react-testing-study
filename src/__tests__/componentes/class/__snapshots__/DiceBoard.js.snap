// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiceBoard 테스트 렌더링을 할수있다 1`] = `
<DiceRoller>
  <div
    className="dice-right-panel"
  >
    <div
      className="dice-result"
    />
    <div
      className="dice-desc"
    >
      <button
        onClick={[Function]}
      >
        던지기
      </button>
    </div>
  </div>
</DiceRoller>
`;
